test/gp/kernel.jl

```diff
@@ -60,6 +60,16 @@ using LinearAlgebra
             differentiable_kernel_tests(Matern52(), ȳ, Ȳ, Ȳ_sq, X0, X1, X2)
         end
 
+        @timedtestset "precomputed" begin
+            X = randn(rng, N, N)
+            K = X * X'
+            k = Linear()
+            @test pw(Precomputed(K), 1:N) ≈ pw(k, ColVecs(X'))
+            @test pw(Precomputed(K), 1:N, 1:N) ≈ pw(k, ColVecs(X'), ColVecs(X'))
+            @test ew(Precomputed(K), 1:N) ≈ ew(k, ColVecs(X'))
+            @test ew(Precomputed(K), 1:N, 1:N) ≈ ew(k, ColVecs(X'), ColVecs(X'))
+        end
```

Comment:
Looks like the `kernel_tests`, as opposed to `differential_kernel_tests` works properly. Something like the following might be better:
```julia
    x = ColVecs(randn(rng, N, N))
    K = pw(linear(), x)
    k = precomputed(K)
    @test pw(k, 1:N, 1:N) == pw(linear(), x)
    kernel_tests(k, 1:N-1, 2:N, 1:N)
```